File: web/src/api/modules/module.ts
import request from '../request'

export interface FeatureModule {
  code: string
  name: string
  description?: string | null
  enabled: boolean
  visible: boolean
}

export const moduleApi = {
  getMyModules() {
    return request.get('/modules')
  },

  listAllModules() {
    return request.get('/modules/all')
  },

  updateModule(code: string, data: { enabled?: boolean; visible?: boolean }) {
    return request.patch(`/modules/${code}`, data)
  }
}

File: server/app/schemas/module.py
from typing import Optional

from pydantic import BaseModel


class ModuleOut(BaseModel):
    code: str
    name: str
    description: Optional[str] = None
    enabled: bool = True
    visible: bool = True


class ModuleUpdate(BaseModel):
    enabled: Optional[bool] = None
    visible: Optional[bool] = None